test/integration/component/test_persistent_networks.py

```diff
@@ -1459,6 +1459,135 @@ def test_delete_account(self):
+            self.apiclient,
+            self.services["isolated_network"],
+            networkofferingid=self.isolated_persistent_network_offering.id,
+            accountid=self.account.name,
+            domainid=self.domain.id,
+            zoneid=self.zone.id)
+
+        response = verifyNetworkState(
+            self.apiclient,
+            network.id,
+            "implemented")
+        exceptionOccured = response[0]
+        isNetworkInDesiredState = response[1]
+        exceptionMessage = response[2]
+
+        if (exceptionOccured or (not isNetworkInDesiredState)):
+            self.fail(exceptionMessage)
+        self.assertIsNotNone(
+            network.vlan,
+            "vlan must not be null for persistent network")
+        try:
+            if self.zone.networktype == "Basic":
+                list_router_response = list_routers(
+                    self.apiclient,
+                    listall="true"
+                )
+            else:
+                list_router_response = list_routers(
+                    self.apiclient,
+                    account=self.account.name,
+                    domainid=self.account.domainid
+                )
+            self.assertEqual(
+                isinstance(list_router_response, list),
+                True,
+                "Check list response returns a valid list"
+            )
+            router = list_router_response[0]
+
+            self.debug("Router ID: %s, state: %s" % (router.id, router.state))
+
+            self.assertEqual(
+                router.state,
+                'Running',
+                "Check list router response for router state"
+            )
+            self.hypervisor = self.testClient.getHypervisorInfo()
+            if self.hypervisor.lower() in ('vmware', 'hyperv'):
+                result = get_process_status(
+                    self.apiclient.connection.mgtSvr,
+                    22,
+                    self.apiclient.connection.user,
+                    self.apiclient.connection.passwd,
+                    router.linklocalip,
+                    "iptables -I INPUT 1 -j DROP",
+                    hypervisor=self.hypervisor
+                )
+            else:
+                try:
+                    hosts = list_hosts(
+                        self.apiclient,
+                        zoneid=router.zoneid,
+                        type='Routing',
+                        state='Up',
+                        id=router.hostid
+                    )
+
+                    self.assertEqual(
+                        isinstance(hosts, list),
+                        True,
+                        "Check list host returns a valid list"
+                    )
+
+                    host = hosts[0]
+                    result = get_process_status(
+                        host.ipaddress,
+                        22, "root", "xenroot", router.linklocalip,
+                        "iptables -I INPUT 1 -j DROP"
+                    )
+
```

Comment:
I believe this is a double hop to router from the host instead of management server. Do you think you can use get_host_credentials for getting host password instead of hard coding?
Otherwise LGTM